_posts/2018-01-01-compatibility-layer.md

```diff
@@ -0,0 +1,113 @@
+---
+layout: post
+title:  "Compatibility Layer"
+subtitle: "0.1 ❤ 0.3"
+author: "Josef Brandl"
+author_github: "MajorBreakfast"
+date:   2018-01-01
+categories: blog
+---
+
+# Futures 0.1 Compatibility Layer
+
+Rust's futures ecosystem is currenlty split in two: On the one hand we have the vibrant ecosystem built around `futures@0.1` with its many libraries working on stable Rust and on the other hand there's the unstable `futures@0.3` ecosystem with support for the ergonomic and powerful `async`/`await` language feature. To bridge the gap between these two worlds we've introduced with `0.3.0-alpha.3` a compatibility layer that makes it possible to use `futures@0.1` and `futures@0.3` futures and streams together in one project. This blog post aims to give an overview over how to use it.
+
+## `Cargo.toml`
+
+The compatibility layer can be enabled by setting the `compat` feature your `Cargo.toml`:
+
+```toml
+futures-preview = { version = "0.3.0-alpha.3", features = ["compat"] }
```

Comment:
Should probably also include the `"tokio-compat"` feature, otherwise the two examples won't run.

Reply:
Done